api/hypershift/v1beta1/azure.go

```diff
@@ -98,6 +98,29 @@ type AzureNodePoolPlatform struct {
 	// If not specified, then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
+
+	// resourceTags is an optional list of additional tags to apply to Azure node
+	// instances. Changes to this field will be propagated in-place to Azure VM instances and their initial OS disks.
+	// Disks created by the storage operator and attached to instances after they are created do not get these tags applied.
+	//
+	// These will be merged with HostedCluster scoped tags, which take precedence in case of conflicts.
+	// These take precedence over tags defined out of band (i.e., tags added manually or by other tools outside of HyperShift) in Azure in case of conflicts.
+	//
+	// See https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support and
+	// https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources for
+	// information on tagging Azure resources. Most Azure resources support up to 50 tags
+	// per resource/resource group/subscription by default. Tag names can be up to 512 characters
+	// (though storage accounts limit tag names to 128 characters), tag values can be up to 256
+	// characters, and empty tag values are allowed. However, certain resource types including
+	// Automation, Content Delivery Network (CDN), Public/Private DNS zones, and Log Analytics
+	// saved searches remain limited to 15 tags total. OpenShift reserves 5 tags for its use,
+	// leaving up to 45 tags available for user use on most resources (10 tags on limited resources).
+	//
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	ResourceTags []AzureResourceTag `json:"resourceTags,omitempty"`
```

Comment:
Will any changes to this field cause the a rollout of new nodes? If so that should be documented.

`// +rollout` should also be added here if so.